test/Microsoft.Framework.Internal.Test/BufferEntryCollectionTest.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Framework.Internal
+{
+    public class BufferEntryCollectionTest
+    {
+        [Fact]
+        public void Add_AddsBufferEntries()
+        {
+            // Arrange
+            var collection = new BufferEntryCollection();
+            var inner = new BufferEntryCollection();
+
+            // Act
+            inner.Add("World");
+            collection.Add("Hello");
+            collection.Add(new[] { 'a', 'b', 'c' }, 1, 2);
+            collection.Add(inner);
```

Comment:
To be more specific about what I ask here: most of the units added in this PR are fine. Except this one. In unit test we want one __Act__ per test so as to easily distinguish source of the error when test fails. Here any one of the method fail will fail the test. It will consume more time to diagnose. Separate the tests into three.